### DISCUSSION TITLE
Compile the modified code.

### DISCUSSION THREAD

**Discussion by chairmanmao256** [mdolab/dafoam#192] [General]
Hello! Every time I modify the codes in dafoam, I type:
```
./Allclean
. $HOME/dafoam/loadDAFoam.sh && \
./Allmake 2> errorLog.txt && \
source $HOME/dafoam/OpenFOAM/OpenFOAM-v1812-ADR/etc/bashrc && \
./Allclean && ./Allmake 2> errorLog.txt && \
source $HOME/dafoam/OpenFOAM/OpenFOAM-v1812-ADF/etc/bashrc && \
./Allclean && ./Allmake 2> errorLog.txt && \
pip install .
```
to compile the dafoam all over again. This process takes me about half an hour no matter how small the modification I've made to the source code. This long compilation really slows down my code development. **I wonder if there's a way to only compile the files that I've modified and the files that are linked with them?** 

Thank you!!!

**Comment by friedenhe** [comment]
Yes, if you make a change, you have to recompile everything. But, if you run `./Allmake incompressible`, it will compile only the incompressible libs and solver, it should be much faster.

**Reply by chairmanmao256** [reply]
Got it! Thank you!

**Reply by hosseinnegahban** [reply]
How could we test our code before recompiling everything?
